### DISCUSSION TITLE
Need Assistance in Retrieving Parent GUIDs for IfcTypeObject Entities in IFC4 File Parsing Scrip and creating Hierarchy as first parent being IFCObject entity than IfcTypeObject and them instance while parsing the IFC 4

### DISCUSSION THREAD

**Discussion by ashi7upt** [IfcOpenShell/IfcOpenShell#3664] [Q&A] [answered]
I'm interested in obtaining the parent GUID for IfcType object entities such as IfcDoorType or IfcSpaceType. Subsequently, I aim to establish a structured hierarchy similar to the following example:

- IfcBuildingStorey
    -IfcSpace
        -IfcSpaceType
             -Name of the Space
             
Could you assist me in achieving this?

**Comment by Moult** [comment]
What do you mean by "parent" GUID? If you mean the "space", types are not contained in spaces.

**Reply by ashi7upt** [reply]
So how do I, club all the IfcTypeObject under their respective IfcObject class for example

![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/86945132/9d1d9f94-7ed1-4d95-90a2-f7a27e1b1541)

**Reply by Moult** [reply, marked as answer]
The building storey is a container of the door. You can get it by doing `ifcopenshell.util.element.get_container(door)`. Alternatively you can get all doors in the container by doing `ifcopenshell.util.element.get_decomposition(storey)`.

The door is a single occurrence (Revit would call it an instance) of the door type. You can get the door type from a single occurrence by doing `ifcopenshell.util.element.get_type(door)`. Inversely, you can get all occurrences of a door type by doing `ifcopenshell.util.element.get_types(door_type)`.
